src/Microsoft.OpenApi.Readers/OpenApiStreamReader.cs

```diff
@@ -102,6 +103,51 @@ public OpenApiDocument Read(Stream input, out OpenApiDiagnostic diagnostic)
             return document;
         }
 
+        /// <summary>
+        /// Reads the stream input and parses the fragment of an OpenAPI descinto an Open API Element.
+        /// </summary>
+        /// <param name="input">Stream containing OpenAPI description to parse.</param>
+        /// <param name="version">Version of the OpenAPI specification that the fragment conforms to.</param>
+        /// <param name="diagnostic">Returns diagnostic object containing errors detected during parsing</param>
+        /// <returns>Instance of newly created OpenApiDocument</returns>
+        public T ReadFragment<T>(Stream input, OpenApiSpecVersion version, out OpenApiDiagnostic diagnostic) where T : IOpenApiElement
+        {
+            ParsingContext context;
+            YamlDocument yamlDocument;
+            diagnostic = new OpenApiDiagnostic();
+
+            // Parse the YAML/JSON
+            try
+            {
+                yamlDocument = LoadYamlDocument(input);
+            }
+            catch (SyntaxErrorException ex)
+            {
+                diagnostic.Errors.Add(new OpenApiReaderError(ex));
+                return default(T);
+            }
+
+            context = new ParsingContext
+            {
+                ExtensionParsers = _settings.ExtensionParsers
+            };
+
+            IOpenApiElement element = null;
+
+            try
+            {
+                // Parse the OpenAPI element
+                element = context.ParseFragment<T>(yamlDocument, version, diagnostic);
+            }
+            catch (OpenApiException ex)
+            {
+                diagnostic.Errors.Add(new OpenApiError(ex));
+            }
+
```

Comment:
Can we also validate the fragment after parsing?

Reply:
@PerthCharern I'm not sure.  I didn't think we could because of references, but as they are unresolved that shouldn't be a problem.  Now I'm not sure.  Will think about it some more.

Reply:
Added it.  Seems to be fine.